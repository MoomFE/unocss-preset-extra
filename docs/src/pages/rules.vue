<template>
  <!-- 侧边栏 -->
  <BasicSidebar ref="sidebarRef" @setDrawerOpenBtn="(fn) => emit('setRenderNavbarLeft', fn)">
    <div class="w-60" un:flex="none">
      <div class="w-60 h-[calc(100%-64px)] fixed scrollbar" un:b-r="1 solid gray op-36">
        <n-menu v-model:value="value" :options="options" />
      </div>
    </div>
    <template #drawer>
      <n-menu v-model:value="value" :options="options" @update:value="sidebarRef.closeDrawer()" />
    </template>
  </BasicSidebar>

  <!-- 内容区域 -->
  <div class="px-6" un:p="x-6 t-5 b-10" un:flex="grow">
    <div class="text-3xl font-bold capitalize mb-1">{{ value }}</div>
    <router-view v-slot="{ Component, route }">
      <transition un:animated="~ faster" enter-active-class="animated-fade-in" leave-active-class="animated-fade-out" mode="out-in">
        <component :is="h('div', Component)" :key="route.path" />
      </transition>
    </router-view>
  </div>
</template>

<script lang="ts" setup>
  import type { MenuProps } from 'naive-ui';
  import { camelCase, upperFirst } from 'lodash';

  const emit = defineEmits(['setRenderNavbarLeft']);

  const route = useRoute();
  const router = useRouter();

  const sidebarRef = ref();

  /** 规则列表 */
  const rules = ['size', 'elevation', 'animated'];
  /** 菜单 */
  const options: MenuProps['options'] = rules.map(key => ({ label: key, key }));
  /** 当前选中的菜单项 */
  const value = ref(
    route.path.split('/').reverse()[0],
  );

  whenever(value, (key) => {
    router.push({
      name: `Rules/${upperFirst(camelCase(key))}`,
    });
  });
</script>

<route lang="yaml">
name: Rules
redirect:
  name: Rules/Size
meta:
  layout: home
  layoutContentClass: px-0!
</route>
