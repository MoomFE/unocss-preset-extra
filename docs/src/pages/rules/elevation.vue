<template>
  <div class="text-sm">
    <OuterChain href="https://vuetifyjs.com/zh-Hans/styles/elevation/" code>Vuetify 海拔</OuterChain> 样式
  </div>
  <n-divider />

  <n-h2>使用</n-h2>
  <DemoCard :code="codeStr">
    <div class="py-12">
      <ElevationDiv :index="index" :opacity="opacity" />
    </div>
    <template #options="{ showCode }">
      <n-form label-placement="left" label-width="4em" :show-feedback="false">
        <client-only>
          <n-form-item label="层级">
            <n-slider v-model:value="index" :max="24" />
          </n-form-item>
          <n-form-item label="透明度">
            <n-slider v-model:value="opacity" :max="100" />
          </n-form-item>
          <n-form-item v-if="showCode" label="简写">
            <n-checkbox v-model:checked="abbr" />
          </n-form-item>
        </client-only>
      </n-form>
    </template>
  </DemoCard>

  <n-h2>所有层级</n-h2>
  <n-card>
    <div un:flex="~ wrap gap-7.5">
      <ElevationDiv v-for="i in 25" :key="i" :index="i - 1" />
    </div>
  </n-card>
</template>

<script lang="tsx" setup>
  /** 层级 */
  const index = ref(6);
  /** 透明度 */
  const opacity = ref(100);
  /** 简写 */
  const abbr = ref(false);

  /** 代码字符串 */
  const codeStr = computed(() => {
    const el = abbr.value ? 'el' : 'elevation';
    const op = abbr.value ? 'op' : 'opacity';
    const opValue = opacity.value;

    return `<div class="${el}-${index.value}${opValue === 100 ? '' : ` ${el}-${op}-${opacity.value}`}" />`;
  });

  /** 海拔 */
  const ElevationDiv = ({ index, opacity }: { index: string | number; opacity?: number }) => (
    <div class={`size-25 flex justify-center items-center rounded c-white bg-sky el-${index} el-op-${opacity ?? 100}`}>{ index }</div>
  );
</script>

<route lang="yaml">
name: Rules/Elevation
</route>
