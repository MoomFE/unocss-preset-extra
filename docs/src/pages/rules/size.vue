<template>
  <div>size</div>
</template>

<script lang="ts" setup>

</script>

<route lang="yaml">
name: Rules/Size
</route>
