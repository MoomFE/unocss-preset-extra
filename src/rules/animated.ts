import { type CSSObject, type Rule } from 'unocss';
import { type Theme } from '@unocss/preset-mini';
import { handler } from '@unocss/preset-mini/utils';
import animatedJSON from './animated.json';

function getAnimated() {
  return animatedJSON as unknown as {
    [key: string]: {
      animationName: string
      css: CSSObject
      keyframes: string
    }
  };
}

export const durationShortcuts = {
  faster: 0.5,
  fast: 0.8,
  slow: 2,
  slower: 3,
};

/**
 * animate.css
 */
export const animatedRules: Rule<Theme>[] = [
  [
    /^animated$/,
    () => {
      return {
        '--une-animated-duration': '1s',
        'animation-duration': 'var(--une-animated-duration)',
        'animation-fill-mode': 'both',
      };
    },
    {
      autocomplete: ['animated'],
    },
  ],
  [
    new RegExp(`^animated-(${Object.keys(animatedJSON).join('|')})$`),
    ([, name]) => {
      const { animationName, css, keyframes } = getAnimated()[name];

      return [
        `@keyframes ${animationName} { ${keyframes} }`,
        css,
      ];
    },
    {
      autocomplete: [
        `animated-(${Object.keys(animatedJSON).join('|')})`,
      ],
    },
  ],
  [
    /^animated-(infinite|(repeat-(infinite|(\d+(\.\d+)?))))$/,
    ([,,, repeat]) => {
      const isInfinite = !repeat || repeat === 'infinite';

      return {
        'animation-iteration-count': isInfinite ? 'infinite' : repeat,
      };
    },
    {
      autocomplete: [
        'animated-infinite',
        'animated-repeat-infinite',
        'animated-repeat-<num>',
      ],
    },
  ],
  [
    /^animated-delay-(none|(\d+(\.\d+)?(m?s)?))$/,
    ([, d]) => ({
      'animation-delay': d === 'none' ? '0ms' : handler.bracket.cssvar.time(d),
    }),
    {
      autocomplete: [
        'animated-delay-none',
        'animated-delay-$duration',
      ],
    },
  ],
  [
    /^animated-(((fast|slow)(?:er)?)|duration-(none|(\d+(\.\d+)?(m?s)?)))/,
    ([_,, shortcut,, v]) => {
      if (shortcut) {
        return {
          'animation-duration': `calc(var(--une-animated-duration) * ${durationShortcuts[shortcut as keyof typeof durationShortcuts]});`,
        };
      }

      return {
        'animation-duration': v === 'none' ? '0ms' : handler.bracket.cssvar.time(v),
      };
    },
    {
      autocomplete: [
        `animated-(${Object.keys(durationShortcuts).join('|')})`,
        'animated-duration-none',
        'animated-duration-$duration',
      ],
    },
  ],
];
