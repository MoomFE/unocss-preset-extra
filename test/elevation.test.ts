import { createGenerator, presetAttributify, presetUno } from 'unocss';
import { describe, expect, test } from 'vitest';
import { createAutocomplete } from '@unocss/autocomplete';
import { omit } from 'lodash-es';
import postcss from 'postcss';
import postcssJs from 'postcss-js';
import { ambient, ambientOpacity, penumbra, penumbraOpacity, umbra, umbraOpacity } from '@@/src/rules/elevation';
import { presetExtra } from '@/index';

function removeUnusedItems(cssJson: object) {
  return omit(cssJson, ['*,::before,::after', '::backdrop']);
}

function createElevationRules(name: string) {
  return Object.assign({}, ...Array.from({ length: 25 }).map((_, index) => {
    return {
      [`.${name}-${index}`]: {
        boxShadow: `${umbra[index]} rgba(0, 0, 0, calc(${umbraOpacity} * var(--une-el-opacity, var(--un-shadow-opacity, 1)))), `
                    + `${penumbra[index]} rgba(0, 0, 0, calc(${penumbraOpacity} * var(--une-el-opacity, var(--un-shadow-opacity, 1)))), `
                    + `${ambient[index]} rgba(0, 0, 0, calc(${ambientOpacity} * var(--une-el-opacity, var(--un-shadow-opacity, 1))))`,
      },
    };
  }));
}

function createElevationOpacityRules(name: string) {
  return Object.assign({}, ...Array.from({ length: 101 }).map((_, i) => {
    return {
      [`.${name}-${i}`]: {
        '--une-el-opacity': `${i / 100}`,
      },
    };
  }));
}

describe('elevation', async () => {
  const generator = createGenerator({
    presets: [
      presetUno(),
      presetAttributify(),
      presetExtra(),
    ],
  });

  const autocomplete = createAutocomplete(generator);

  test('(shadow-)?(el|elevation)-*', async () => {
    // el-*
    const { css } = await generator.generate(
      Array.from({ length: 36 }).map((_, i) => `el-${i}`).join(' '), // 多生成几个, 测试是否非 0 ~ 24 的 elevation 会被忽略
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css)),
      ),
    ).toEqual(
      createElevationRules('el'),
    );

    // elevation-*
    const { css: css2 } = await generator.generate(
      Array.from({ length: 36 }).map((_, i) => `elevation-${i}`).join(' '), // 多生成几个, 测试是否非 0 ~ 24 的 elevation 会被忽略
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css2)),
      ),
    ).toEqual(
      createElevationRules('elevation'),
    );

    // shadow-el-*
    const { css: css3 } = await generator.generate(
      Array.from({ length: 36 }).map((_, i) => `shadow-el-${i}`).join(' '), // 多生成几个, 测试是否非 0 ~ 24 的 elevation 会被忽略
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css3)),
      ),
    ).toEqual(
      createElevationRules('shadow-el'),
    );

    // shadow-elevation-*
    const { css: css4 } = await generator.generate(
      Array.from({ length: 36 }).map((_, i) => `shadow-elevation-${i}`).join(' '), // 多生成几个, 测试是否非 0 ~ 24 的 elevation 会被忽略
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css4)),
      ),
    ).toEqual(
      createElevationRules('shadow-elevation'),
    );
  });

  test('(shadow-)?(el|elevation)-(op|opacity)-*', async () => {
    // el-op-*
    const { css } = await generator.generate(
      Array.from({ length: 101 }).map((_, i) => `el-op-${i}`).join(' '),
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css)),
      ),
    ).toEqual(
      createElevationOpacityRules('el-op'),
    );

    // el-opacity-*
    const { css: css2 } = await generator.generate(
      Array.from({ length: 101 }).map((_, i) => `el-opacity-${i}`).join(' '),
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css2)),
      ),
    ).toEqual(
      createElevationOpacityRules('el-opacity'),
    );

    // elevation-op-*
    const { css: css3 } = await generator.generate(
      Array.from({ length: 101 }).map((_, i) => `elevation-op-${i}`).join(' '),
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css3)),
      ),
    ).toEqual(
      createElevationOpacityRules('elevation-op'),
    );

    // elevation-opacity-*
    const { css: css4 } = await generator.generate(
      Array.from({ length: 101 }).map((_, i) => `elevation-opacity-${i}`).join(' '),
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css4)),
      ),
    ).toEqual(
      createElevationOpacityRules('elevation-opacity'),
    );

    // shadow-el-op-*
    const { css: css5 } = await generator.generate(
      Array.from({ length: 101 }).map((_, i) => `shadow-el-op-${i}`).join(' '),
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css5)),
      ),
    ).toEqual(
      createElevationOpacityRules('shadow-el-op'),
    );

    // shadow-el-opacity-*
    const { css: css6 } = await generator.generate(
      Array.from({ length: 101 }).map((_, i) => `shadow-el-opacity-${i}`).join(' '),
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css6)),
      ),
    ).toEqual(
      createElevationOpacityRules('shadow-el-opacity'),
    );

    // shadow-elevation-op-*
    const { css: css7 } = await generator.generate(
      Array.from({ length: 101 }).map((_, i) => `shadow-elevation-op-${i}`).join(' '),
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css7)),
      ),
    ).toEqual(
      createElevationOpacityRules('shadow-elevation-op'),
    );

    // shadow-elevation-opacity-*
    const { css: css8 } = await generator.generate(
      Array.from({ length: 101 }).map((_, i) => `shadow-elevation-opacity-${i}`).join(' '),
    );

    expect(
      removeUnusedItems(
        postcssJs.objectify(postcss.parse(css8)),
      ),
    ).toEqual(
      createElevationOpacityRules('shadow-elevation-opacity'),
    );
  });

  test('autocomplete', async () => {
    expect(
      await autocomplete.suggest('el-'),
    ).toMatchSnapshot();

    expect(
      await autocomplete.suggest('elevation-'),
    ).toMatchSnapshot();

    expect(
      await autocomplete.suggest('shadow-elevation-'),
    ).toMatchSnapshot();
  });
});
