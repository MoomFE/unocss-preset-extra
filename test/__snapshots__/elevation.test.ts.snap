// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`elevation > autocomplete 1`] = `
[
  "el-op-0",
  "el-op-1",
  "el-op-2",
  "el-op-3",
  "el-op-4",
  "el-op-5",
  "el-op-6",
  "el-op-7",
  "el-op-8",
  "el-op-9",
  "el-op-10",
  "el-op-11",
  "el-op-12",
  "el-op-13",
  "el-op-14",
  "el-op-15",
  "el-op-16",
  "el-op-17",
  "el-op-18",
  "el-op-19",
  "el-op-20",
  "el-op-21",
  "el-op-22",
  "el-op-23",
  "el-op-24",
  "el-op-25",
  "el-op-26",
  "el-op-27",
  "el-op-28",
  "el-op-29",
  "el-op-30",
  "el-op-31",
  "el-op-32",
  "el-op-33",
  "el-op-34",
  "el-op-35",
  "el-op-36",
  "el-op-37",
  "el-op-38",
  "el-op-39",
  "el-op-40",
  "el-op-41",
  "el-op-42",
  "el-op-43",
  "el-op-44",
  "el-op-45",
  "el-op-46",
  "el-op-47",
  "el-op-48",
  "el-op-49",
  "el-op-50",
  "el-op-51",
  "el-op-52",
  "el-op-53",
  "el-op-54",
  "el-op-55",
  "el-op-56",
  "el-op-57",
  "el-op-58",
  "el-op-59",
  "el-op-60",
  "el-op-61",
  "el-op-62",
  "el-op-63",
  "el-op-64",
  "el-op-65",
  "el-op-66",
  "el-op-67",
  "el-op-68",
  "el-op-69",
  "el-op-70",
  "el-op-71",
  "el-op-72",
  "el-op-73",
  "el-op-74",
  "el-op-75",
  "el-op-76",
  "el-op-77",
  "el-op-78",
  "el-op-79",
  "el-op-80",
  "el-op-81",
  "el-op-82",
  "el-op-83",
  "el-op-84",
  "el-op-85",
  "el-op-86",
  "el-op-87",
  "el-op-88",
  "el-op-89",
  "el-op-90",
  "el-op-91",
  "el-op-92",
  "el-op-93",
  "el-op-94",
  "el-op-95",
  "el-op-96",
  "el-op-97",
  "el-op-98",
  "el-op-99",
  "el-op-100",
  "el-opacity-0",
  "el-opacity-1",
  "el-opacity-2",
  "el-opacity-3",
  "el-opacity-4",
  "el-opacity-5",
  "el-opacity-6",
  "el-opacity-7",
  "el-opacity-8",
  "el-opacity-9",
  "el-opacity-10",
  "el-opacity-11",
  "el-opacity-12",
  "el-opacity-13",
  "el-opacity-14",
  "el-opacity-15",
  "el-opacity-16",
  "el-opacity-17",
  "el-opacity-18",
  "el-opacity-19",
  "el-opacity-20",
  "el-opacity-21",
  "el-opacity-22",
  "el-opacity-23",
  "el-opacity-24",
  "el-opacity-25",
  "el-opacity-26",
  "el-opacity-27",
  "el-opacity-28",
  "el-opacity-29",
  "el-opacity-30",
  "el-opacity-31",
  "el-opacity-32",
  "el-opacity-33",
  "el-opacity-34",
  "el-opacity-35",
  "el-opacity-36",
  "el-opacity-37",
  "el-opacity-38",
  "el-opacity-39",
  "el-opacity-40",
  "el-opacity-41",
  "el-opacity-42",
  "el-opacity-43",
  "el-opacity-44",
  "el-opacity-45",
  "el-opacity-46",
  "el-opacity-47",
  "el-opacity-48",
  "el-opacity-49",
  "el-opacity-50",
  "el-opacity-51",
  "el-opacity-52",
  "el-opacity-53",
  "el-opacity-54",
  "el-opacity-55",
  "el-opacity-56",
  "el-opacity-57",
  "el-opacity-58",
  "el-opacity-59",
  "el-opacity-60",
  "el-opacity-61",
  "el-opacity-62",
  "el-opacity-63",
  "el-opacity-64",
  "el-opacity-65",
  "el-opacity-66",
  "el-opacity-67",
  "el-opacity-68",
  "el-opacity-69",
  "el-opacity-70",
  "el-opacity-71",
  "el-opacity-72",
  "el-opacity-73",
  "el-opacity-74",
  "el-opacity-75",
  "el-opacity-76",
  "el-opacity-77",
  "el-opacity-78",
  "el-opacity-79",
  "el-opacity-80",
  "el-opacity-81",
  "el-opacity-82",
  "el-opacity-83",
  "el-opacity-84",
  "el-opacity-85",
  "el-opacity-86",
  "el-opacity-87",
  "el-opacity-88",
  "el-opacity-89",
  "el-opacity-90",
  "el-opacity-91",
  "el-opacity-92",
  "el-opacity-93",
  "el-opacity-94",
  "el-opacity-95",
  "el-opacity-96",
  "el-opacity-97",
  "el-opacity-98",
  "el-opacity-99",
  "el-opacity-100",
  "el-0",
  "el-1",
  "el-2",
  "el-3",
  "el-4",
  "el-5",
  "el-6",
  "el-7",
  "el-8",
  "el-9",
  "el-10",
  "el-11",
  "el-12",
  "el-13",
  "el-14",
  "el-15",
  "el-16",
  "el-17",
  "el-18",
  "el-19",
  "el-20",
  "el-21",
  "el-22",
  "el-23",
  "el-24",
]
`;

exports[`elevation > autocomplete 2`] = `
[
  "elevation-op-0",
  "elevation-op-1",
  "elevation-op-2",
  "elevation-op-3",
  "elevation-op-4",
  "elevation-op-5",
  "elevation-op-6",
  "elevation-op-7",
  "elevation-op-8",
  "elevation-op-9",
  "elevation-op-10",
  "elevation-op-11",
  "elevation-op-12",
  "elevation-op-13",
  "elevation-op-14",
  "elevation-op-15",
  "elevation-op-16",
  "elevation-op-17",
  "elevation-op-18",
  "elevation-op-19",
  "elevation-op-20",
  "elevation-op-21",
  "elevation-op-22",
  "elevation-op-23",
  "elevation-op-24",
  "elevation-op-25",
  "elevation-op-26",
  "elevation-op-27",
  "elevation-op-28",
  "elevation-op-29",
  "elevation-op-30",
  "elevation-op-31",
  "elevation-op-32",
  "elevation-op-33",
  "elevation-op-34",
  "elevation-op-35",
  "elevation-op-36",
  "elevation-op-37",
  "elevation-op-38",
  "elevation-op-39",
  "elevation-op-40",
  "elevation-op-41",
  "elevation-op-42",
  "elevation-op-43",
  "elevation-op-44",
  "elevation-op-45",
  "elevation-op-46",
  "elevation-op-47",
  "elevation-op-48",
  "elevation-op-49",
  "elevation-op-50",
  "elevation-op-51",
  "elevation-op-52",
  "elevation-op-53",
  "elevation-op-54",
  "elevation-op-55",
  "elevation-op-56",
  "elevation-op-57",
  "elevation-op-58",
  "elevation-op-59",
  "elevation-op-60",
  "elevation-op-61",
  "elevation-op-62",
  "elevation-op-63",
  "elevation-op-64",
  "elevation-op-65",
  "elevation-op-66",
  "elevation-op-67",
  "elevation-op-68",
  "elevation-op-69",
  "elevation-op-70",
  "elevation-op-71",
  "elevation-op-72",
  "elevation-op-73",
  "elevation-op-74",
  "elevation-op-75",
  "elevation-op-76",
  "elevation-op-77",
  "elevation-op-78",
  "elevation-op-79",
  "elevation-op-80",
  "elevation-op-81",
  "elevation-op-82",
  "elevation-op-83",
  "elevation-op-84",
  "elevation-op-85",
  "elevation-op-86",
  "elevation-op-87",
  "elevation-op-88",
  "elevation-op-89",
  "elevation-op-90",
  "elevation-op-91",
  "elevation-op-92",
  "elevation-op-93",
  "elevation-op-94",
  "elevation-op-95",
  "elevation-op-96",
  "elevation-op-97",
  "elevation-op-98",
  "elevation-op-99",
  "elevation-op-100",
  "elevation-opacity-0",
  "elevation-opacity-1",
  "elevation-opacity-2",
  "elevation-opacity-3",
  "elevation-opacity-4",
  "elevation-opacity-5",
  "elevation-opacity-6",
  "elevation-opacity-7",
  "elevation-opacity-8",
  "elevation-opacity-9",
  "elevation-opacity-10",
  "elevation-opacity-11",
  "elevation-opacity-12",
  "elevation-opacity-13",
  "elevation-opacity-14",
  "elevation-opacity-15",
  "elevation-opacity-16",
  "elevation-opacity-17",
  "elevation-opacity-18",
  "elevation-opacity-19",
  "elevation-opacity-20",
  "elevation-opacity-21",
  "elevation-opacity-22",
  "elevation-opacity-23",
  "elevation-opacity-24",
  "elevation-opacity-25",
  "elevation-opacity-26",
  "elevation-opacity-27",
  "elevation-opacity-28",
  "elevation-opacity-29",
  "elevation-opacity-30",
  "elevation-opacity-31",
  "elevation-opacity-32",
  "elevation-opacity-33",
  "elevation-opacity-34",
  "elevation-opacity-35",
  "elevation-opacity-36",
  "elevation-opacity-37",
  "elevation-opacity-38",
  "elevation-opacity-39",
  "elevation-opacity-40",
  "elevation-opacity-41",
  "elevation-opacity-42",
  "elevation-opacity-43",
  "elevation-opacity-44",
  "elevation-opacity-45",
  "elevation-opacity-46",
  "elevation-opacity-47",
  "elevation-opacity-48",
  "elevation-opacity-49",
  "elevation-opacity-50",
  "elevation-opacity-51",
  "elevation-opacity-52",
  "elevation-opacity-53",
  "elevation-opacity-54",
  "elevation-opacity-55",
  "elevation-opacity-56",
  "elevation-opacity-57",
  "elevation-opacity-58",
  "elevation-opacity-59",
  "elevation-opacity-60",
  "elevation-opacity-61",
  "elevation-opacity-62",
  "elevation-opacity-63",
  "elevation-opacity-64",
  "elevation-opacity-65",
  "elevation-opacity-66",
  "elevation-opacity-67",
  "elevation-opacity-68",
  "elevation-opacity-69",
  "elevation-opacity-70",
  "elevation-opacity-71",
  "elevation-opacity-72",
  "elevation-opacity-73",
  "elevation-opacity-74",
  "elevation-opacity-75",
  "elevation-opacity-76",
  "elevation-opacity-77",
  "elevation-opacity-78",
  "elevation-opacity-79",
  "elevation-opacity-80",
  "elevation-opacity-81",
  "elevation-opacity-82",
  "elevation-opacity-83",
  "elevation-opacity-84",
  "elevation-opacity-85",
  "elevation-opacity-86",
  "elevation-opacity-87",
  "elevation-opacity-88",
  "elevation-opacity-89",
  "elevation-opacity-90",
  "elevation-opacity-91",
  "elevation-opacity-92",
  "elevation-opacity-93",
  "elevation-opacity-94",
  "elevation-opacity-95",
  "elevation-opacity-96",
  "elevation-opacity-97",
  "elevation-opacity-98",
  "elevation-opacity-99",
  "elevation-opacity-100",
  "elevation-0",
  "elevation-1",
  "elevation-2",
  "elevation-3",
  "elevation-4",
  "elevation-5",
  "elevation-6",
  "elevation-7",
  "elevation-8",
  "elevation-9",
  "elevation-10",
  "elevation-11",
  "elevation-12",
  "elevation-13",
  "elevation-14",
  "elevation-15",
  "elevation-16",
  "elevation-17",
  "elevation-18",
  "elevation-19",
  "elevation-20",
  "elevation-21",
  "elevation-22",
  "elevation-23",
  "elevation-24",
]
`;

exports[`elevation > autocomplete 3`] = `
[
  "shadow-elevation-op-0",
  "shadow-elevation-op-1",
  "shadow-elevation-op-2",
  "shadow-elevation-op-3",
  "shadow-elevation-op-4",
  "shadow-elevation-op-5",
  "shadow-elevation-op-6",
  "shadow-elevation-op-7",
  "shadow-elevation-op-8",
  "shadow-elevation-op-9",
  "shadow-elevation-op-10",
  "shadow-elevation-op-11",
  "shadow-elevation-op-12",
  "shadow-elevation-op-13",
  "shadow-elevation-op-14",
  "shadow-elevation-op-15",
  "shadow-elevation-op-16",
  "shadow-elevation-op-17",
  "shadow-elevation-op-18",
  "shadow-elevation-op-19",
  "shadow-elevation-op-20",
  "shadow-elevation-op-21",
  "shadow-elevation-op-22",
  "shadow-elevation-op-23",
  "shadow-elevation-op-24",
  "shadow-elevation-op-25",
  "shadow-elevation-op-26",
  "shadow-elevation-op-27",
  "shadow-elevation-op-28",
  "shadow-elevation-op-29",
  "shadow-elevation-op-30",
  "shadow-elevation-op-31",
  "shadow-elevation-op-32",
  "shadow-elevation-op-33",
  "shadow-elevation-op-34",
  "shadow-elevation-op-35",
  "shadow-elevation-op-36",
  "shadow-elevation-op-37",
  "shadow-elevation-op-38",
  "shadow-elevation-op-39",
  "shadow-elevation-op-40",
  "shadow-elevation-op-41",
  "shadow-elevation-op-42",
  "shadow-elevation-op-43",
  "shadow-elevation-op-44",
  "shadow-elevation-op-45",
  "shadow-elevation-op-46",
  "shadow-elevation-op-47",
  "shadow-elevation-op-48",
  "shadow-elevation-op-49",
  "shadow-elevation-op-50",
  "shadow-elevation-op-51",
  "shadow-elevation-op-52",
  "shadow-elevation-op-53",
  "shadow-elevation-op-54",
  "shadow-elevation-op-55",
  "shadow-elevation-op-56",
  "shadow-elevation-op-57",
  "shadow-elevation-op-58",
  "shadow-elevation-op-59",
  "shadow-elevation-op-60",
  "shadow-elevation-op-61",
  "shadow-elevation-op-62",
  "shadow-elevation-op-63",
  "shadow-elevation-op-64",
  "shadow-elevation-op-65",
  "shadow-elevation-op-66",
  "shadow-elevation-op-67",
  "shadow-elevation-op-68",
  "shadow-elevation-op-69",
  "shadow-elevation-op-70",
  "shadow-elevation-op-71",
  "shadow-elevation-op-72",
  "shadow-elevation-op-73",
  "shadow-elevation-op-74",
  "shadow-elevation-op-75",
  "shadow-elevation-op-76",
  "shadow-elevation-op-77",
  "shadow-elevation-op-78",
  "shadow-elevation-op-79",
  "shadow-elevation-op-80",
  "shadow-elevation-op-81",
  "shadow-elevation-op-82",
  "shadow-elevation-op-83",
  "shadow-elevation-op-84",
  "shadow-elevation-op-85",
  "shadow-elevation-op-86",
  "shadow-elevation-op-87",
  "shadow-elevation-op-88",
  "shadow-elevation-op-89",
  "shadow-elevation-op-90",
  "shadow-elevation-op-91",
  "shadow-elevation-op-92",
  "shadow-elevation-op-93",
  "shadow-elevation-op-94",
  "shadow-elevation-op-95",
  "shadow-elevation-op-96",
  "shadow-elevation-op-97",
  "shadow-elevation-op-98",
  "shadow-elevation-op-99",
  "shadow-elevation-op-100",
  "shadow-elevation-opacity-0",
  "shadow-elevation-opacity-1",
  "shadow-elevation-opacity-2",
  "shadow-elevation-opacity-3",
  "shadow-elevation-opacity-4",
  "shadow-elevation-opacity-5",
  "shadow-elevation-opacity-6",
  "shadow-elevation-opacity-7",
  "shadow-elevation-opacity-8",
  "shadow-elevation-opacity-9",
  "shadow-elevation-opacity-10",
  "shadow-elevation-opacity-11",
  "shadow-elevation-opacity-12",
  "shadow-elevation-opacity-13",
  "shadow-elevation-opacity-14",
  "shadow-elevation-opacity-15",
  "shadow-elevation-opacity-16",
  "shadow-elevation-opacity-17",
  "shadow-elevation-opacity-18",
  "shadow-elevation-opacity-19",
  "shadow-elevation-opacity-20",
  "shadow-elevation-opacity-21",
  "shadow-elevation-opacity-22",
  "shadow-elevation-opacity-23",
  "shadow-elevation-opacity-24",
  "shadow-elevation-opacity-25",
  "shadow-elevation-opacity-26",
  "shadow-elevation-opacity-27",
  "shadow-elevation-opacity-28",
  "shadow-elevation-opacity-29",
  "shadow-elevation-opacity-30",
  "shadow-elevation-opacity-31",
  "shadow-elevation-opacity-32",
  "shadow-elevation-opacity-33",
  "shadow-elevation-opacity-34",
  "shadow-elevation-opacity-35",
  "shadow-elevation-opacity-36",
  "shadow-elevation-opacity-37",
  "shadow-elevation-opacity-38",
  "shadow-elevation-opacity-39",
  "shadow-elevation-opacity-40",
  "shadow-elevation-opacity-41",
  "shadow-elevation-opacity-42",
  "shadow-elevation-opacity-43",
  "shadow-elevation-opacity-44",
  "shadow-elevation-opacity-45",
  "shadow-elevation-opacity-46",
  "shadow-elevation-opacity-47",
  "shadow-elevation-opacity-48",
  "shadow-elevation-opacity-49",
  "shadow-elevation-opacity-50",
  "shadow-elevation-opacity-51",
  "shadow-elevation-opacity-52",
  "shadow-elevation-opacity-53",
  "shadow-elevation-opacity-54",
  "shadow-elevation-opacity-55",
  "shadow-elevation-opacity-56",
  "shadow-elevation-opacity-57",
  "shadow-elevation-opacity-58",
  "shadow-elevation-opacity-59",
  "shadow-elevation-opacity-60",
  "shadow-elevation-opacity-61",
  "shadow-elevation-opacity-62",
  "shadow-elevation-opacity-63",
  "shadow-elevation-opacity-64",
  "shadow-elevation-opacity-65",
  "shadow-elevation-opacity-66",
  "shadow-elevation-opacity-67",
  "shadow-elevation-opacity-68",
  "shadow-elevation-opacity-69",
  "shadow-elevation-opacity-70",
  "shadow-elevation-opacity-71",
  "shadow-elevation-opacity-72",
  "shadow-elevation-opacity-73",
  "shadow-elevation-opacity-74",
  "shadow-elevation-opacity-75",
  "shadow-elevation-opacity-76",
  "shadow-elevation-opacity-77",
  "shadow-elevation-opacity-78",
  "shadow-elevation-opacity-79",
  "shadow-elevation-opacity-80",
  "shadow-elevation-opacity-81",
  "shadow-elevation-opacity-82",
  "shadow-elevation-opacity-83",
  "shadow-elevation-opacity-84",
  "shadow-elevation-opacity-85",
  "shadow-elevation-opacity-86",
  "shadow-elevation-opacity-87",
  "shadow-elevation-opacity-88",
  "shadow-elevation-opacity-89",
  "shadow-elevation-opacity-90",
  "shadow-elevation-opacity-91",
  "shadow-elevation-opacity-92",
  "shadow-elevation-opacity-93",
  "shadow-elevation-opacity-94",
  "shadow-elevation-opacity-95",
  "shadow-elevation-opacity-96",
  "shadow-elevation-opacity-97",
  "shadow-elevation-opacity-98",
  "shadow-elevation-opacity-99",
  "shadow-elevation-opacity-100",
  "shadow-elevation-0",
  "shadow-elevation-1",
  "shadow-elevation-2",
  "shadow-elevation-3",
  "shadow-elevation-4",
  "shadow-elevation-5",
  "shadow-elevation-6",
  "shadow-elevation-7",
  "shadow-elevation-8",
  "shadow-elevation-9",
  "shadow-elevation-10",
  "shadow-elevation-11",
  "shadow-elevation-12",
  "shadow-elevation-13",
  "shadow-elevation-14",
  "shadow-elevation-15",
  "shadow-elevation-16",
  "shadow-elevation-17",
  "shadow-elevation-18",
  "shadow-elevation-19",
  "shadow-elevation-20",
  "shadow-elevation-21",
  "shadow-elevation-22",
  "shadow-elevation-23",
  "shadow-elevation-24",
]
`;
